### PR TITLE
as a end user I would like to be able to attach ETH/tokens to my status update on gitcoin, so i can share money

### PR DESCRIPTION
**Pull request by zoek1**
##### Description

- [X] a 'attach tokens' button on the status update bottom
- [X]  when its clicked, a user is prompted to select a token (any token listed at /settings/tokens can be added) and enter an amount
image

- [x]  user can submit the status update and when they do this function ( https://github.com/gitcoinco/web/blob/master/app/assets/v2/js/pages/bulk_payout.js#L140-L155 ) is called and they are prompted to send tokens. once they send tokens, the status update is posted
DeepinScreenshot_20200131071339

- [X]  on the status update, there is a nice visual box that shows there are tokens attached to the post.
- [x]  after users comment on the post, the poster (and only the poster) can click an "award" button next to each comment to award any commenter the tokens.
image

##### Refers/Fixes

 #5881

### CODE REVIEW

**Comment by zoek1** [comment]
I'm based on changes of #5883, is the UI fine to specify the amount and award the tokens?

**Comment by danlipert** [comment]
Excited for this! Let me know once its ready for review

**Comment by zoek1** [comment]
I'm asking to solve some questions in the original issue to unlock this :smiley:

**Comment by owocki** [comment]
> I'm asking to solve some questions in the original issue to unlock this 😃

i wanna push on this soon..  r u still blocked? link me the questions?

**Comment by zoek1** [comment]
No problem @owocki, in working on this right now!   In the morning the PR will ve ready to Review :thumbs_up:

**Comment by zoek1** [comment]
It's ready for review @owocki @danlipert :smiley:

**Comment by owocki** [comment]
feedback - 
1. wow it works end to end.. yay! 
2. https://bits.owocki.com/geurNd5Y can we make this area black, and also show the token amount + token name  eg "0.01 ETH attached" isntead of' Tip attached"
3. when i award the tip, does the to user get an email?
4. when i visit the page incognito i get a 500 https://bits.owocki.com/qGuDm7PO
5. when i post a new comment i get this issue https://bits.owocki.com/yAuLEK8J

**Comment by zoek1** [comment]
- [X] show the token amount + token name eg "0.01 ETH attached" isntead of' Tip attached"
- [X] when i award the tip, does the to user get an email?
- [X] when i visit the page incognito i get a 500 https://bits.owocki.com/qGuDm7PO
- [X] when i post a new comment i get this issue https://bits.owocki.com/yAuLEK8J

> https://bits.owocki.com/geurNd5Y can we make this area black

i'm not sure what region should be black, The button, the comments or the status: :sweat_smile: 

@owocki

**Comment by zoek1** [comment]
@owocki fix conflicts done!

**Comment by zoek1** [comment]
@owocki I recently read the tip functionality is live, is this PR still relevant?

**Comment by octavioamu** [comment]
I will say yes, still totally valid and interesting feature.

**Comment by owocki** [comment]
yeah staking on posts is an important mechanism.. would love to get this working on the new townsquare + get it into an upcoming release...

**Comment by zoek1** [comment]
Do you consider change something based on what you tested? I'll integrate this weekend the new square town into this. 
@owocki

**Comment by danlipert** [comment]
Hey @zoek1 - @owocki is Out-of-office for a bit so I'll take over in the meantime, looks like this is rebased and ready for review and merge, right?

**Comment by zoek1** [comment]
Yep @danlipert, it's ready!

**Comment by octavioamu** [comment]
Seems the merge of giphy bring some conflicts here

**Comment by zoek1** [comment]
I'll check and fix the conflicts 🙂

**Comment by zoek1** [comment]
Fixed! :smile:  @octavioamu

**Comment by zoek1** [comment]
Update branch :smile:

**Comment by zoek1** [comment]
@octavioamu @owocki can i get some feedback on this?

**Comment by danlipert** [comment]
I just gave this a test and it seemed to work pretty well, but has some small visual issues - @zoek1 could you give this one last check if you have a chance - thanks!

**Comment by zoek1** [comment]
Sure @danlipert, I'll ping you when I fixed the UI issues. 😀 could you point some visual issues here? I appreciate that a lot!

**Comment by danlipert** [comment]
@zoek1 i noticed the form when you attach tokens looks a bit off. Also, I think there might be an issue with the comments count - maybe thats just my local though. Also looks like there are some conflicts

**Comment by zoek1** [comment]
@danlipert the conflicts are fixed, also i fixed the visual issues when attach comments.

**Comment by androolloyd** [comment]
@zoek1 the conflicts that exist right now seem pretty minor to adjust, can you rebase your changes onto master so we can move this along?

**Comment by thelostone-mc** [comment]
@zoek1 ^
we can get this in post that

**Comment by thelostone-mc** [comment]
@zoek1 ^ would you wanna close this out and open a fresh one later with everything resolved when you get the time ?

**Comment by zoek1** [comment]
Sure no problem, only one question. I know the gitcoin core team is pretty busy also I don't want to resolve conflicts again and again because that could generate noise in the git history (in this PR the half of the commits is for fix conflicts) and in the review. I think if you give me a date  i'll have this PR clean and ready for the given date  :pray: 

@thelostone-mc

**Comment by thelostone-mc** [comment]
We can get this in tom at the earliest if the conflicts are resolved ! If you need more time => before next wed works too

**Comment by zoek1** [comment]
before next wed works for me! :)

File: app/townsquare/templates/townsquare/index.html [hunk 1]
```diff
@@ -182,6 +182,18 @@
...
+    <script src="{% static "v2/js/lib/ipfs-api.js" %}"></script>
+    <script src="{% static "v2/js/ipfs.js" %}"></script>
+    <script src="{% static "v2/js/abi.js" %}"></script>
+    <script src="{% static "v2/js/amounts.js" %}"></script>
+    <script src="{% static "v2/js/lib/secrets.min.js" %}"></script>
+    <script src="{% static "v2/js/ethereumjs-accounts.js" %}"></script>
+    <script src="/dynamic/js/tokens_dynamic.js"></script>
+    <script src="{% static "onepager/js/send.js" %}"></script>
+
+    <script src="{% static "v2/js/search.js" %}"></script>
```
Comment (octavioamu) [inline app/townsquare/templates/townsquare/index.html:196]: This is loaded in the footer scripts today, can we rebase this pr with the last changes and ping me back?
Also review if some of the others files are already lodade into footer_scripts.html

Comment (zoek1) [reply app/townsquare/templates/townsquare/index.html:196]: Ok, give 5 minutes

Comment (zoek1) [reply app/townsquare/templates/townsquare/index.html]: I double-checked and only 'search.js' was both, here and in footer_scripts

File: app/dashboard/templates/profiles/status_box.html [hunk 2]
```diff
@@ -61,6 +63,28 @@
...
+
+              </div>
+          </div>
+          <div class="col-md-4">
+            <div class="form-group">
+              <label class="sup featured-text">Token</label>
+              <select class="form-control" id="attachToken">
+                <option value="0x0">ETH</option>
+              {% for token in TOKENS %}
+                <option value="{{ token.token_addres }}">{{ token.token_name }}</option>
```
Comment (danlipert) [inline app/dashboard/templates/profiles/status_box.html:81]: Was this only tested with ETH? There is a typo here for the token_address

Comment (zoek1) [reply app/dashboard/templates/profiles/status_box.html]: Thanks, I missed that. I tested with some ERC20 tokens like MANA and GNT. 
image

**Comment by zoek1** [comment]
@danlipert do you think this could be merge soon? it's tested with eth and tokens :stuck_out_tongue: 

@owocki

**Pull request merged**